cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java

```diff
@@ -164,9 +180,9 @@ public void validateTags(Id.NamespacedId entityId, String ... tags) throws Inval
 
   // Validate the key should not be be reserved {@link BusinessMetadataDataset.TAGS_KEY}
   private void validateTagReservedKey(Id.NamespacedId entityId, String key) throws InvalidMetadataException {
-    if (BusinessMetadataDataset.TAGS_KEY.equals(key.toLowerCase())) {
+    if (MetadataDataset.TAGS_KEY.equals(key.toLowerCase())) {
```

Comment:
Hmmm, not sure what that jira will describe. I think we can handle it during the implementation of system metadata. Or do you think a jira with the title "Response of listing all properties or all tags for an entity should make it easy to distinguish between system and user metadata" makes sense?